File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {};

DnaTranscriber.prototype.toRna = function(dna) {
  var rna = '';
  //iterate over the dna strand, check each nucleotides and include the correspondent nucleotide in the rna strand
  for (var i = 0; i < dna.length; i++) {
    switch (dna.charAt(i)) {
      case "C":
        rna += "G";
        break;
      case "G":
        rna += "C";
        break;
      case "A":
        rna += "U";
        break;
      case "T":
        rna += "A";
        break;
    }
  }
  return rna;
}

module.exports = DnaTranscriber;

File: c/rna-transcription/rna_transcription.c
#include "rna_transcription.h"

char *to_rna(const char *dna)
{
    
    if (dna == NULL) {
        char *empty = (char *) calloc(1, sizeof(char));
        return empty;
    }

    size_t size_dna = strlen(dna);

    char *rna = (char *) calloc(size_dna + 1, sizeof(char));
    if (rna == NULL) {
        return NULL;
    }

    for (size_t i = 0; i < size_dna; i++) {
        switch (dna[i]) {
            case 'G': rna[i] = 'C'; break;
            case 'C': rna[i] = 'G'; break;
            case 'T': rna[i] = 'A'; break;
            case 'A': rna[i] = 'U'; break;
        }
    }

    return rna;
}